bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/types/ParameterizedType.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.runtime.api.types;
+
+/**
+ * {@code ParameterizedType} represents the parameterized type in dependently-typed functions.
+ *
+ * @since 2.0.0
+ */
+public interface ParameterizedType extends Type {
```

Comment:
Do we need to export creating ParameterizedType using Typecreator api for interop developers?

Reply:
Right now, I can't think of a scenario where there'll be a requirement to create one in the external implementation. 

`ParameterizedType` is only expected to be used as the return type of a function/method atm anyway. Given that we don't expose creating a `FunctionType` right now, as discussed I think we can defer this until there is a requirement.